### ISSUE TITLE
Better version of IIR filter

### ISSUE THREAD

**Issue by p0nce** [AuburnSounds/Dplug#431]
Create a new IIR filter capacility, splitted in two.
- [ ] Pole-Zero realization should be separated from computation of those Pole and Zero placement.
- [ ] Smoothing should be bu. The PoleZero computation should be push-style, and no pole-zero pair inversion occur without its control.
- [ ] DC auto-initialization (inversion) should be supported because it's very useful for control signals
- [ ] multi-channel should be supported, since it's probably more efficient
- [ ] should allow multiple Pole-Zero realization (templated ?)

**Summing Up**

The role of the pole-zero **realization** is to filter, hold state, hold smoothing state. It has a maximum number of pole-zero (8 ? 6 ?)
The role of the pole-zero **computation** is to compute pole and zeros when needed.  

Computation feeds realization with pole and zero, and choose their relative order.

Advantages: the last filter structure you need.

**Comment by p0nce** [comment]
Desired realizations: Direct Form I / transposed Directo Form II / Andy Simper's oversampled state variable filter

**Comment by p0nce** [comment]
First need to make a few dplug:audio "processors", like a silence detector or a delayline

**Comment by p0nce** [comment]
Doing this necessitates in turn to define AudioContext and audio processors in a future-proof way
